pkg/util/kubernetes/kubernetes.go

```diff
@@ -199,7 +198,6 @@ func NewNatsPodSpec(clusterName string, cs spec.ClusterSpec, owner metav1.OwnerR
 
 	container := natsPodContainer(clusterName, cs.Version)
 	container = containerWithLivenessProbe(container, natsLivenessProbe(cs.TLS.IsSecureClient()))
-	container = containerWithReadinessProbe(container, natsReadinessProbe(clusterName))
```

Comment:
So no readiness probe? This means any NATS pod, regardless of its state, will be resolvable and proxied to, no matter its state, which may result in bad client UX.

Reply:
right now yes, though we could revisit this part later and have something like a parameter in the server `/routez` to assert that the cluster is formed for the case when the pod is warming up.